src/components/sidebar/index.tsx

```diff
@@ -22,49 +22,51 @@ export interface MenuItem {
   path?: string
   routes?: MenuItem[]
   title?: string
+  /* Temporary solution to allow rendering of unlinked headings, as in designs */
+  heading?: string
+  /* Temporary solution to show external link icon, for demo purposes */
+  _demoExternalLink?: boolean
 }
 
 interface SidebarProps {
   menuItems: MenuItem[]
+  /** Optional { text, url } to use for the "← Back to..." link at the top of the sidebar */
+  backToLink?: {
+    text: string
+    url: string
+  }
 }
 
 const addItemMetadata = (
   currentPath: string,
   items: MenuItem[]
 ): { foundActiveItem: boolean; itemsWithMetadata: MenuItem[] } => {
-  const currentPathSplit = currentPath.split('/')
-  const currentProductSlug = currentPathSplit[1]
-  const currentProductSubpage = currentPathSplit[2]
-
   let foundActiveItem = false
 
   const itemsWithMetadata = items.map((item) => {
     const itemCopy = { ...item }
 
-    if (item.divider) {
-      return itemCopy
-    }
-
-    if (foundActiveItem) {
-      itemCopy[item.routes ? 'hasActiveChild' : 'isActive'] = false
-    }
-
     if (item.routes) {
       const result = addItemMetadata(currentPath, item.routes)
-      foundActiveItem = result.foundActiveItem
       itemCopy.routes = result.itemsWithMetadata
-      itemCopy.hasActiveChild = result.foundActiveItem
+      // Note: if an active item has already been found,
+      // we do not flag this category as active.
+      itemCopy.hasActiveChild = !foundActiveItem && result.foundActiveItem
       itemCopy.id = `submenu-${itemCopy.title
         .replace(/( |\.)/g, '-')
         .replace(/-+/g, '-')
         .toLowerCase()}`
+      // Flag if we've found an active item
+      foundActiveItem = itemCopy.hasActiveChild || foundActiveItem
     } else if (item.path) {
-      foundActiveItem = currentPath.endsWith(item.path)
-      itemCopy.isActive = foundActiveItem
-      itemCopy.fullPath = `/${currentProductSlug}/${currentProductSubpage}/${item.path}`
+      // Note: if an active item has already been found,
+      // we do not flag this node as active.
+      itemCopy.isActive = !foundActiveItem && currentPath.endsWith(item.path)
       itemCopy.id = `menu-item-${itemCopy.fullPath
         .replace(/\//g, '-')
         .toLowerCase()}`.replace(/-+/g, '-')
+      // Flag if we've found an active item
+      foundActiveItem = itemCopy.isActive || foundActiveItem
```

Comment:
Can we talk about why this is being changed so much? Is something broken?

Reply:
@ashleemboyer  For sure, apologies for not reviewing this yet, was planning on it. There are two main changes I'm suggesting here:

1. Removing the `fullPath` creation, and instead having this be the responsibility of the consumer of `sidebar`. This lets us fill in the `fullPath` values only once, at build time rather than on the client. More importantly, this enables the flexibility we need to show custom, non-external `sidebar` items on the `/waypoint` product view. i don't necessarily mean to suggest this as a long-term solution (though I think it might be a viable path!), mainly just focused on getting the product view ready to demo. Without this change, I don't think we can properly show the links needed on the `/waypoint` page sidebar (ie would need to use `href` but that would show external link icons which would not match design... or we could pass down `basePath` as suggested earlier, but I think that might add complexity compared to this solution).
2. The `foundActiveItem` refactoring is less important, happy to drop these changes... but I believe the `if (foundActiveItem)` conditional might not be functioning as expected... it seems to set an item's `hasActiveChild` or `isActive` to `false` in that first `if` block, but then in the subsequent `if / else` block, I think that would be undone by the `itemCopy.hasActiveChild` or `itemCopy.isActive` assignments. Refactored because after making the `fullPath` changes, for some reason I was having issues getting the category items to expand. I think longer-term, we may want to leverage something like the [existing utility in docs-sidenav](https://github.com/hashicorp/react-components/blob/main/packages/docs-sidenav/utils/flag-active-nodes.js) to retain existing behaviour of also highlighting `href` values, and handling some rare "alias" edge cases.

Reply:
☝️ Let me know if that's helpful - happy to adjust the approach here 🙇‍♂️

Reply:
I'd like to have non-textual discussion about this since there are a few different things at play, but looks like it's working right now so no blocker here for merging. 👍🏻